### PR TITLE
Implement Entity and DAO layer

### PR DESCRIPTION
**Pull request by TeamoYOOOO**
This PR includes the implementation of Entity and DAO layer.

Some concerns to discuss:
- Enum or Entity, which one is better for category? Currently I have it as Enum but I also include a Category class in the entity package (it's not used). Implement category as an entity seem to be more complicated. 
- Do we need another DTO layer to avoid exposing sensitive entity objects such as password? We don't have billing in our design for now but it would still be great to not to expose our design pattern to the front end. 
- I used Hibernate annotation such as ”@OnetoMany“ for dabatase. Not sure if this would cause tight coupling. I think we can keep this for now and modify later if needed. 

Next step:
- Custom Exceptions for error handling (e.g. CustomerNotExistException)
- Unit test for custom SQL query. This can be done after database connection being set up.
- Set up the connection between AWS RDS and the application. I have already created the database. Need to add the info to application.properties and set up an environment variable for sensitive info such as password. 
- Implementation of service layer
- Unit tests for methods in service layer.

### CODE REVIEW

File: src/main/java/com/sysops/dao/ArticleDao.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.sysops.entity.Article;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+
+import java.util.List;
+
+public interface ArticleDao extends CrudRepository<Article, Long> {
+    @Query("SELECT article FROM Article a WHERE a.title LIKE %:keyword% OR a.content LIKE %:keyword%")
+    List<Article> findByKeyword(String keyword);
```
Comment (TeamoYOOOO) [inline src/main/java/com/sysops/dao/ArticleDao.java]: I forgot to add "@Param("keyword")" to the parameter. Will include it in later commit.

**Review by kumashi7 — APPROVED** [review]
Nice start! For the concerns:
1. I'd say for now Enum is good for category. Have the feeling that we won't insert any rows to category table.
2. Password is a good advice. May save that for phase 2 implementation?
3. For now Hibernate is good to me.

A lot of good ideas and we can use then in phase 2 (more like a improvement or scale things up)

**Comment by yiyuzhang0604** [comment]
Great start! 

I know that this PR is only for entity and DAO layer, but just wondering if there is any chance we can test to see if these work?

**Review by yiyuzhang0604 — APPROVED** [review]

**Comment by TeamoYOOOO** [comment]
> 

All dao interface extend from CRUDRepository so I think we are good for that part. The ArticleDao has one custom sql query so I will write a unit test for that method. I would like to seperate test and production database, so I will use separate configuration files for the testing environment and use the in memory database H2 for testing. Still trying to figure out how to do that.

**Comment by TeamoYOOOO** [comment]
> Great start!
> 
> I know that this PR is only for entity and DAO layer, but just wondering if there is any chance we can test to see if these work?

All dao interface extend from CRUDRepository so I think we are good for that part. The ArticleDao has one custom sql query so I will write a unit test for that method. I would like to seperate test and production database, so I will use separate configuration files for the testing environment and use the in memory database H2 for testing. Still trying to figure out how to do that.

**Comment by TeamoYOOOO** [comment]
implementation of entity and dao layer

**Pull request merged**